sdcard/tools/wifi_start.sh

```diff
@@ -114,30 +118,4 @@ start()
     fi
 }
 
-start_valhalla()
```

Comment:
I've deleted this otherwise the boot sequence will loop, first it will launch valhalla.sh wich call wifi_start.sh wich call valhalla.sh and again and again. I'm sure ther is a better way to do this, like maybe implementing a thinner verson of wifi_start.sh inside valhalla.sh.

Reply:
flow changed after the version 148 and finally fixed after 158.

Before (still is in 0099),  /etc/init.d/rcS-> check SDCARD and set flag to use custom boot.
S50gm with flag, call custom ft_boot.sh inside SDCARD, which calls valhalla.sh

Due to the change in the firmware (xiaomi after 148). Custom boot ft_boos.sh was tricky to make work and most of the time failed so the flow changed to. (this is done when upgrading under pos_ota.sh)

/etc/init.d/rcS -> S41wifi -> /mnt/data/ot_wifi_tool/wifi_start.sh (custom file done at upgrading through OTA). -> Valhalla.sh

wifi_start.sh is modified to call valhalla and add a protection for looping. You might take a look to the wifi_start.sh inside the release under the tool folder.

#Loop protection.
[ -f /tmp/valhalla.pid ] && return
	echo 1 > /tmp/valhalla.pid

This code: 
## in Mode 3, the wifi is configured in AP. Restart Wifi in default mode (0).
echo 0 > /tmp/ft_mode
/etc/init.d/S41wifi restart

was there in the past for something different, now it should be to reconfigure the wifi.